queue/mock_test.go

```diff
@@ -66,25 +71,51 @@ func newMockRemoteQueue(opts mockRemoteQueueOptions) (*mockRemoteQueue, error) {
 	}
 	mq := &mockRemoteQueue{remoteQueue: opts.queue}
 	if opts.driver != nil {
-		if err := opts.queue.SetDriver(opts.driver); err != nil {
+		if err := mq.SetDriver(opts.driver); err != nil {
 			return nil, errors.Wrap(err, "configuring queue with driver")
 		}
 	}
 	if opts.makeDispatcher != nil {
-		dispatcher := opts.makeDispatcher(mq)
+		mq.dispatcher = opts.makeDispatcher(mq)
```

Comment:
Does it make sense to make this even if there is not driver? In other words, should the above line be `if opts.makeDispatcher != nil && opts.driver != nil`?

Reply:
I set the mock queue to initialize a default dispatcher if it's unset, since the driver can't realistically operate without a dispatcher.